pkg/apis/core/validation/validation.go

```diff
@@ -3069,6 +3069,28 @@ func ValidateNodeSelector(nodeSelector *core.NodeSelector, fldPath *field.Path)
 	return allErrs
 }
 
+// validateTopologySelectorLabelRequirement tests that the specified TopologySelectorLabelRequirement fields has valid data
+func validateTopologySelectorLabelRequirement(rq core.TopologySelectorLabelRequirement, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(rq.Values) == 0 {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("values"), "must specify as least one value"))
+	}
+	allErrs = append(allErrs, unversionedvalidation.ValidateLabelName(rq.Key, fldPath.Child("key"))...)
+
+	return allErrs
+}
+
+// ValidateTopologySelectorTerm tests that the specified topology selector term has valid data
+func ValidateTopologySelectorTerm(term core.TopologySelectorTerm, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
```

Comment:
Let's also add a feature gate check here while it's alpha.

Reply:
You mean check whether feature DynamicProvisioningScheduling is enabled here?
But as @saad-ali suggested, the struct might not be only used by this feature, and I think we should have already validate related feature in higher level who make use of the struct (AllowedTopologies for DynamicProvisioningScheduling).

Reply:
Yes, I think in the future, other components can use it, but while we are still developing it as an alpha feature, let's feature gate it.

Reply:
added